csharp/src/Microsoft.ML.OnnxRuntime/NativeMethods.shared.cs

```diff
@@ -10,6 +10,7 @@ namespace Microsoft.ML.OnnxRuntime
     public struct OrtApiBase
     {
         public IntPtr GetApi;
+        public IntPtr GetTrainingApi;
```

Comment:
>public IntPtr GetTrainingApi; [](http://example.com/codeflow?start=8&length=29)

Can we make it a separate source file? The struct here is very long already.

Reply:
Also, it is probably mandatory to make sure when you deal with inference only scenarios, non of the training things are invoked and require training functionality.

Reply:
done